File: ContentManager.Api/Presentation/Controllers/ContentPostController.cs
using ContentManager.Api.Contracts.Application.Models.Requests;
using ContentManager.Api.Contracts.Application.Services;
using Filebin.Shared.Domain.Models;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace ContentManager.Api.Presentation.Controllers;

[Authorize]
[Route("api/posts")]
[ApiController]
public class ContentPostController(IContentPostService service) : ControllerBase {
    [HttpGet]
    [AllowAnonymous]
    public async Task<IActionResult> GetPage([FromQuery] PageDesc pageDesc) {
        return Ok(await service.GetPageAsync(pageDesc));
    }

    [HttpGet("/count")]
    [AllowAnonymous]
    public async Task<IActionResult> GetCount() {
        return Ok(await service.GetCountAsync());
    }

    [HttpGet("{id:guid}")]
    [AllowAnonymous]
    public async Task<IActionResult> GetPost([FromRoute] Guid id) {
        return Ok(await service.GetByIdAsync(id));
    }

    [HttpPost]
    public async Task<IActionResult> CreatePost([FromBody] ContentPostCreateRequest createRequest) {
        return Ok(await service.CreateAsync(createRequest));
    }

    [HttpPut("{id:guid}")]
    public async Task<IActionResult> UpdatePost([FromRoute] Guid id, [FromBody] ContentPostUpdateRequest updateRequest) {
        await service.UpdateAsync(id, updateRequest);
        return Ok();
    }

    [HttpDelete("{id:guid}")]
    public async Task<IActionResult> DeletePost([FromRoute] Guid id) {
        await service.DeleteAsync(id);
        return Ok();
    }
}

File: content-manager-frontend/src/lib/api/content-post-controller.ts
import axios from "axios";

import config from '@/config.json';
import type {
  ContentPostCreateRequest,
  ContentPostResponse,
  ContentPostUpdateRequest,
  PageDesc
} from "./models";

const BASE_URL = `${config.apiUrl}/api/posts`;


export const getPage = async (pageDesc: PageDesc) => {
  const response = await axios.get<ContentPostResponse[]>(BASE_URL, { params: pageDesc });
  return response.data;
};

export const getCount = async () => {
  const response = await axios.get<number>(`${BASE_URL}/count`);
  return response.data;
};

export const getPost = async (id: string) => {
  const response = await axios.get<ContentPostResponse>(`${BASE_URL}/${id}`);
  return response.data;
};

export const createPost = async (createRequest: ContentPostCreateRequest) => {
  const response = await axios.post<string>(BASE_URL, createRequest);
  return response.data;
};

export const updatePost = async (id: string, updateRequest: ContentPostUpdateRequest) => {
  await axios.put(`${BASE_URL}/${id}`, updateRequest);
};

export const deletePost = async (id: string) => {
  await axios.delete(`${BASE_URL}/${id}`);
};